line_profiler/autoprofile/util_import.py

```diff
@@ -0,0 +1,476 @@
+"""
+extract the below functions from packages ubelt & xdoctest using liberator used in profmod_extractor.py
+https://github.com/Kitware/liberator
+https://github.com/Erotemic/xdoctest (v1.0.1)
+https://github.com/Erotemic/ubelt (v1.1.2)
+
+ubelt.util_import.modname_to_modpath
+ubelt.util_import.modpath_to_modname
+xdoctest.static_analysis.package_modpaths
+"""
+
+from os.path import (abspath, basename, dirname, exists, expanduser, isdir,
+                     isfile, join, realpath, relpath, split, splitext)
+import os
+import sys
+
+"""
+ubelt.util_import.modname_to_modpath
+ubelt.util_import.modpath_to_modname
+"""
+def _extension_module_tags():
+    """
+    Returns valid tags an extension module might have
+
+    Returns:
+        List[str]
+    """
+    import sysconfig
+    tags = []
+    # handle PEP 3149 -- ABI version tagged .so files
+    # ABI = application binary interface
+    tags.append(sysconfig.get_config_var('SOABI'))
+    tags.append('abi3')  # not sure why this one is valid but it is
+    tags = [t for t in tags if t]
```

Comment:
This could be simplified and sped up for longer lists with [itertools.compress](https://docs.python.org/3/library/itertools.html#itertools.compress)

Reply:
This is autogenerated code ported from my [ubelt](https://github.com/Erotemic/ubelt) library. I will take care of ensuring we have a clean port for this code.

I doubt the list ever gets long enough where it matters, but I [benchmarked it](https://github.com/Erotemic/misc/blob/main/tests/python/bench_compress.py) and I was surprised to learn how performant `list(it.compress(items, items))` is versus `[x for x in items if x]`. Always nice to pick up small tricks like this.

![image](https://user-images.githubusercontent.com/3186211/218524017-45b05c84-2090-4588-a564-2265548fbf64.png)

However, when there are only a few items (< 7) using a comprehension does seem faster: 
![image](https://user-images.githubusercontent.com/3186211/218524463-a23b9d8e-bde4-4ea6-b323-03709f140ccb.png)